dist/gcc-compatible/Hacl_SHA3.h

```diff
@@ -35,16 +35,6 @@ extern "C" {
 #include "krml/lowstar_endianness.h"
 #include "krml/internal/target.h"
 
-#include "Lib_Memzero0.h"
-
-extern const uint32_t Hacl_Impl_SHA3_keccak_rotc[24U];
-
-extern const uint32_t Hacl_Impl_SHA3_keccak_piln[24U];
-
-extern const uint64_t Hacl_Impl_SHA3_keccak_rndc[24U];
-
-uint64_t Hacl_Impl_SHA3_rotl(uint64_t a, uint32_t b);
-
 void Hacl_Impl_SHA3_state_permute(uint64_t *s);
```

Comment:
that file is not bundled behind an interface (see my comment in the other PR as to why), so you probably would benefit from adding `private` qualifiers in state_permute, loadState and storeState

Reply:
I think permute is called from the streaming APIs and it complains when I turn it private. I guess it would be better to make these functions private and instead expose "absorb_inner" which functions as "update-1-block" for SHA3?

Reply:
curious why streaming calls these functions... wondering if something is inlined too aggressively

Reply:
yeah so the reason is because streaming calls absorb_inner to do the loop over a multiple of a number of blocks, and that one is inline_for_extraction noextract, which means the implementation of SHA3 is replicated in the streaming version and hence generates cross-translation unit calls to permute

my suggested fix is to remove inline_for_extraction noextract from `absorb_inner`, and wait for my other PR to land so that you won't have a performance hit (because as streaming_sha3 and sha3 are merged together into a single file, `absorb_inner` will be marked as `static` and hopefully will happily optimized just as well)

alternatively, you could do `let absorb_inner_non_inline = absorb_inner`, then call `absorb_inner_non_inline` from `Hacl.Hash.SHA3`, which will remove the copy of `absorb_inner` inlined everything in Hacl.Streaming.SHA3, and will allow you to mark permute as private